core/src/fr/hammons/slinc/Tools.scala

```diff
@@ -27,16 +27,20 @@ object Tools:
 
   def sendResourceToCache(name: String): Unit =
     Files.createDirectories(appDataStore)
-    val cacheLocation = appDataStore.resolve(s"$name.c")
-
+    val cachedSharedLibLocation = appDataStore.resolve(s"$name$sharedLibSuffix")
+    val cachedNativeSrcLocation = appDataStore.resolve(s"$name.c")
+    if !Files.exists(cachedSharedLibLocation) then
+      val stream = getClass().getResourceAsStream(s"/native/$name$sharedLibSuffix")
+      if stream != null then Files.copy(stream,cachedSharedLibLocation)
+      return
     if !Files.exists(appDataStore.resolve(s"$name.c")) then
```

Comment:
I don't think it is reasonable to bundle `*.c` or `*.cpp` files in resources and compile them at runtime.

Reply:
I guess current implementation is just for testing, isn't it?

Reply:
So, it's helpful for testing, but I was hoping to have this feature as a last resort for new platforms. It's not portable no, but the .c or .cpp files are supposed to be used as the last resort, and are ignored if there are .so or .dll files with the same name available. See below.